hedera-node/src/main/java/com/hedera/services/store/contracts/ContractsStore.java

```diff
@@ -170,17 +173,24 @@ public void commit() {
 				final var account = ledger.get(accId);
 				ledger.adjustBalance(accId, evmAccount.getBalance().toLong() - account.getBalance());
 			} else {
-				var pair = this.provisionalAccountCreations.get(accId);
-				ledger.create(pair.getKey(), accId, evmAccount.getBalance().toLong(), pair.getValue());
+				if (this.provisionalAccountCreations.containsKey(accId)) {
```

Comment:
I am wondering - why do we iterate through `provisionalAccountCreations` and create it in the ledger only through the `provisionalAccountUpdates`? Is it not better to:

1. Have a separate `forEach` on `provisionalAccountCreations`
2. Add assert that the contract is `smart_contract`
3. Create it

And after that we can iterate through `provisionalAccountUpdates`. 

I think that the current implementation might be bugs if we are creating an account but it is not part of the `provisionalAccountUpdates`. In that case, we won't create the contract at all?